usr/local/sbin/restic_backup.sh

```diff
@@ -15,39 +20,36 @@ exit_hook() {
 }
 trap exit_hook INT TERM
 
-# How many backups to keep.
-RETENTION_DAYS=14
-RETENTION_WEEKS=16
-RETENTION_MONTHS=18
-RETENTION_YEARS=3
-
-# What to backup, and what to not
-BACKUP_PATHS="/ /boot /home"
-# Example below of how dynamically add a path that is mounted e.g. external USB dis.
-[ -d /mnt/media ] && BACKUP_PATHS+=" /mnt/media"
-
-# Set up exclude files: global + path-specific ones.
-BACKUP_EXCLUDES="--exclude-file /etc/restic/backup_exclude"
-for dir in /home/*
-do
-	if [ -f "$dir/.backup_exclude" ]
-	then
-		BACKUP_EXCLUDES+=" --exclude-file $dir/.backup_exclude"
+# Set up exclude files: global + path-specific ones + home directoreis.
+# NOTE that restic will fail the backup if not all listed --exclude-files exist. Thus we should only list them if they are really all available.
+##  Global backup configuration.
+exclusion_args="--exclude-file /etc/restic/backup_exclude"
+## Self-contained backup files per backup path. E.g. having an USB disk at /mnt/media in BACKUP_PATHS, then it can have a /mnt/media/.backup_exclude
+for backup_path in ${BACKUP_PATHS[@]}; do
+	if [ -f "$backup_path/.backup_exclude" ]; then
+		exclusion_args+=" --exclude-file $backup_path/.backup_exclude"
 	fi
 done
-
-BACKUP_TAG=systemd.timer
-
-
-# Set all environment variables like
-# B2_ACCOUNT_ID, B2_ACCOUNT_KEY, RESTIC_REPOSITORY etc.
-source /etc/restic/b2_env.sh
-
-# How many network connections to set up to B2. Default is 5.
-B2_CONNECTIONS=50
+## Additonal service: if /home is being backed up, allow user specifci backup files in home directory or default $XDG_CONFIG_HOME path
+backup_files_from_homedirs() {
+	local homeroot="$1"
+	local args=
+	if [[ $BACKUP_PATHS == *"$homeroot"* ]]; then
+		for homedir in $homeroot/*; do
+			excl_home="$homedir/.backup_exclude"
+			excl_xdg="$homedir/.config/restic/backup_exclude"
+			test -f $excl_home && args+=" --exclude-file $excl_home"
+			test -f $excl_xdg && args+=" --exclude-file $excl_xdg"
+		done
+	fi
+	echo $args
+}
+exclusion_args="$exclusion_args $(backup_files_from_homedirs /home)"
+## And the same service of macOS users, having /Users instead of /home
+exclusion_args="$exclusion_args $(backup_files_from_homedirs /Users)"
```

Comment:
Accidentally it will have a leading space if it returns anything, else empty string, so it would work, if this contract stays true. I guess this is fine, both works :)